### PR TITLE
Framework: Skip Photon for blob and data images in safeImageUrl

### PR DESCRIPTION
**Pull request by aduth**
This pull request primarily seeks to return `blob:` and `data:image/` URLs unmodified if passed to `safeImageUrl`, since these are used for either inline images or in-progress media uploads.

It also generally refactors the module, upgrading to ES6 and adding a README example.

__Testing instructions:__

Verify that all Mocha tests pass as expected:

```
npm run test-client client/lib/safe-image-url
```

### CODE REVIEW

File: client/lib/safe-image-url/test/index.js [hunk 1]
```diff
@@ -1,68 +1,78 @@
...
-		expect( safeImage( 'https://example.com/foo.jpg?bar' ) ).to.be.null;
-		expect( safeImage( 'https://example.com/foo.jpeg?bar' ) ).to.be.null;
-		expect( safeImage( 'https://example.com/foo.gif?bar' ) ).to.be.null;
-		expect( safeImage( 'https://example.com/foo.png?bar' ) ).to.be.null;
+	it( 'should return null for urls with querystrings', () => {
+		expect( safeImageUrl( 'https://example.com/foo?bar' ) ).to.be.null;
+		expect( safeImageUrl( 'https://example.com/foo.jpg?bar' ) ).to.be.null;
+		expect( safeImageUrl( 'https://example.com/foo.jpeg?bar' ) ).to.be.null;
+		expect( safeImageUrl( 'https://example.com/foo.gif?bar' ) ).to.be.null;
+		expect( safeImageUrl( 'https://example.com/foo.png?bar' ) ).to.be.null;
```
Comment (blowery) [inline client/lib/safe-image-url/test/index.js]: we should probably add a few tests with weird protocols too, especially `javascript:`, maybe `about:` and `config:`? Doesn't have to be this PR.

File: client/lib/safe-image-url/index.js [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
+ * @type {Array}
+ */
+const REGEX_EXEMPT_URL = /^(\/(?!\/)|data:image\/[^;]+;|blob:)/;
+
+/**
+ * Pattern matching Automattic-controlled hostnames
+ *
+ * @type {RegExp}
  */
+const REGEXP_A8C_HOST = /^([-a-zA-Z0-9_]+\.)*(gravatar\.com|wordpress\.com|wp\.com|a8c\.com)$/;
```
Comment (blowery) [inline client/lib/safe-image-url/index.js]: yay for escaping the `.`s

could we add a test that exploits the old `.` hole?

**Comment by aduth** [comment]
Added a few more tests for input variations in 97a9339. Will squash and merge shortly after a final round of testing.

**Comment by aduth** [comment]
Changed behavior in f0e44c0 to allow only Blob URLs for current origin. This is more a "better safe than sorry" revision given that Blob URLs, [by spec](https://url.spec.whatwg.org/#local-scheme), are local.

**Pull request merged**